server/app/services/geo/esri/EsriClient.java

```diff
@@ -0,0 +1,88 @@
+package services.geo.esri;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import play.libs.ws.*;
+import services.geo.AddressAttributes;
+import services.geo.AddressLocation;
+
+public class EsriClient implements WSBodyReadables, WSBodyWritables {
+  public static final String ESRI_CONTENT_TYPE = "application/json";
+  public static final String ESRI_FIND_ADDRESS_CANDIDATES_OUT_FIELDS =
+      "Address, SubAddr, City, Region, Postal";
+  public static final String ESRI_FIND_ADDRESS_CANDIDATES_FORMAT = "pjson";
+
+  public final String ESRI_FIND_ADDRESS_CANDIDATES_URL;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_LABELS;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_VALUES;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_URLS;
+  public final ConfigList ESRI_ADDRESS_VERIFICATION_PATHS;
+
+  private final WSClient ws;
```

Comment:
suggestion (non-blocking): 
```suggestion
  private final WSClient wsClient;
```